src/main/kotlin/com/openlattice/chronicle/jobs/DeleteChronicleUsageDataTask.kt

```diff
@@ -0,0 +1,140 @@
+import com.openlattice.chronicle.storage.RedshiftColumns.Companion.STUDY_ID
+import com.openlattice.chronicle.storage.RedshiftDataTables.Companion.CHRONICLE_USAGE_EVENTS
+import com.openlattice.chronicle.storage.StorageResolver
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+/**
+ * @author Solomon Tang <solomon@openlattice.com>
+ */
+class DeleteChronicleUsageDataTask(
+    private val jobService: JobService,
+    private val storageResolver: StorageResolver,
+    private val auditingManager: AuditingManager,
+): Runnable {
+    companion object {
+        private val logger = LoggerFactory.getLogger(DeleteChronicleUsageDataTask::class.java)!!
+
+        /*
+          1. status
+          2. updated_at
+         */
+        private val GET_NEXT_JOB_COLUMNS = listOf(
+            UPDATED_AT,
+            STATUS
+        ).joinToString(",") { it.name }
+
+        private val GET_NEXT_JOB_SQL = """
+            UPDATE ${JOBS.name}
+            SET (${GET_NEXT_JOB_COLUMNS}) = (?, ?)
+            WHERE ${JOB_ID.name} = (
+                SELECT ${JOB_ID.name}
+                FROM ${JOBS.name}
+                WHERE ${JOB_DATA.name} ->> '@type' = '${DeleteStudyUsageData::class.java.name}'
+                AND ${STATUS.name} = '${JobStatus.PENDING.name}'
+                ORDER BY ${CREATED_AT.name} ASC
+                FOR UPDATE SKIP LOCKED
+                LIMIT 1
+            )
+            RETURNING *
+        """.trimIndent()
+
+        private val DELETE_CHRONICLE_STUDY_USAGE_DATA_SQL = """
+            DELETE FROM ${CHRONICLE_USAGE_EVENTS.name}
+            WHERE ${STUDY_ID.name} = ?
+        """.trimIndent()
+
+        private  val UPDATE_FINISHED_JOB_COLUMNS = listOf(
+            UPDATED_AT,
+            STATUS,
+            DELETED_ROWS
+        ).joinToString(",") { it.name }
+
+        private val UPDATE_FINISHED_DELETE_JOB_SQL = """
+            UPDATE ${JOBS.name}
+            SET (${UPDATE_FINISHED_JOB_COLUMNS}) = (?, ?, ?)
+            WHERE ${JOB_ID.name} = ?
+        """.trimIndent()
+    }
+
+    override fun run() {
+        deleteChronicleUsageData()
+    }
+
+    private fun deleteChronicleUsageData() {
+        val hds = storageResolver.getPlatformStorage()
+        AuditedOperationBuilder<ChronicleJob>(hds.connection, auditingManager)
+            .operation { connection ->
+                // pull job from queue with skip locked
+                // deserialize the jobData
+                val job = getNextAvailableJob(connection)
+                if (job.jobData is DeleteStudyUsageData) {
+                    val deletedRows = deleteChronicleStudyUsageData(job.jobData)
+                    // update jobData to include deletedRows
+                    updateFinishedDeleteJob(connection, job.id, deletedRows)
+                }
+                else {
+                    logger.info("No pending DeleteStudyUsageData jobs available")
+                }
+                return@operation job
```

Comment:
You can declare return type as Unit instead of ChronicleJob if you aren't returning job.

Reply:
the grabbed job is being used in the AuditableEvent just below